praxis-isncsci/ui
# [Cell input] UI controls for extra input for `*` values

Closes #117 

[Figma](https://www.figma.com/file/82mMuohRV0zPWnZbZ5upup/isncsci-app?type=design&node-id=480%3A30941&mode=design&t=91fy3ZsMoC16sT14-1)

## Problem

We need to add the input controls to capture extra information when using the star ( * ) flag. This section should be hidden when the star ( * ) flag is not selected.

<img alt="Figma designs for mobile and larger layouts" width="862" src="https://github.com/praxis-isncsci/ui/assets/1294355/a9d60b5a-c105-43f4-ac97-052cc3982a06" />

## Solution

- Added the input controls to collect the extra information
- Added the styles to handle the layout for mobile and larger screens
- Used and exposed design tokens and CSS variables to support future customization of the component

## Testing

- [x] 1. Input controls show on mobile layout

<details>
  <summary>Test case</summary>

  1. Navigate to [ISNCSCI Input Sensory Storybook story](https://64f8d7c6e093108e99084a70-swthpckmmt.chromatic.com/?path=/story/webcomponents-isncsci-input--sensory)
  2. Change the preview size to **large mobile**
  3. Set the `showStarInput` control to `true`
  4. Confirm the input controls are presented according to the designs in Figma 

</details>

---

- [x] 2. Input controls show on larger layout

<details>
  <summary>Test case</summary>

  1. Navigate to [ISNCSCI Input Sensory Storybook story](https://64f8d7c6e093108e99084a70-swthpckmmt.chromatic.com/?path=/story/webcomponents-isncsci-input--sensory)
  2. Reset the preview size
  3. Set the `showStarInput` control to `true`
  4. Confirm the input controls are presented according to the designs in Figma 

</details>

---
